ReactiveCocoa/Swift/Property.swift

```diff
@@ -350,9 +362,28 @@ extension PropertyProtocol {
 	}
 }
 
-/// A read-only, type-erased view of a property.
-public struct AnyProperty<Value>: PropertyProtocol {
-	private let sources: [Any]
+/// A read-only property that can be observed for its changes over time. There are
+/// three categories of read-only property:
+///
+/// # Constant property
+/// Created by `Property(value:)`, the producer and signal of a constant
+/// property would complete immediately when it is initialized.
+///
+/// # Existential property
+/// Created by `Property(_:)`, an existential property passes through the
+/// behavior of the wrapped property.
+///
+/// # Composed property
+/// Created by either the compositional operators in `PropertyProtocol`, or
+/// `Property(initial:followingBy:)`, a composed property presents a
+/// composed view of its source, which can be a set of properties,
+/// a producer, or a signal.
+///
+/// A composed property respects the lifetime of its source rather than its own.
+/// In other words, its producer and signal can outlive the property itself, if
+/// its source outlives it too.
+public final class Property<Value>: PropertyProtocol {
+	public let sources: [AnyObject]?
```

Comment:
Making it non-public would require runtime casting, i.e. what is currently in `RAC5`, since we do not have private conformance in Swift.